pkg/logging/klog_bridge.go

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package logging
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"log/slog"
+	"regexp"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+var klogErrorOverrides = []logLevelOverride{
+	{
+		// TODO: We can drop the misspelled case here once client-go version is bumped to include:
+		//	https://github.com/kubernetes/client-go/commit/ae43527480ee9d8750fbcde3d403363873fd3d89
+		matcher:     regexp.MustCompile("Failed to update lock (optimitically|optimistically).*falling back to slow path"),
+		targetLevel: slog.LevelInfo,
+	},
+}
+
+func initializeKLog() error {
```

Comment:
Might make sense to avoid `DefaultSlogLogger` in cases where we can, e.g. here we can pass it down from `SetupLogging`